src/main/kotlin/stringcalculator/domain/Calculator.kt

```diff
@@ -0,0 +1,27 @@
+package stringcalculator.domain
+
+import java.util.function.BinaryOperator
+
+enum class Calculator(private val sign: String, private val operation: BinaryOperator<Int>) {
```

Comment:
`함수형 인터페이스` 활용 잘하셨네요! 👍👍🤩 

여기서 조금 더 코틀린 스타일로 변경해보면 아래와 같이 변경해볼 수 있을 것 같아요.  🙂
그러면 Java 라이브러리 의존성도 제거할 수 있어요. 😉

```suggestion
enum class Calculator(private val sign: String, private val operation: (Int, Int) -> Int) {
```

Reply:
`ADD`,`SUBSTRACT`, `MULTIPLY`, `DIVIDE`(= enum의 요소들) 모두 객체이고 enum 클래스는 이러한 공통의 성질을 지닌 객체들을 가지게 되는데, 저는 `+-/*`의 공통적인 성질을 표현하는데 Calculator라는 네이밍은 적합하지 않다고 생각해요. 🤔
용주님은 어떻게 생각하시나요? 😃

[enum을 잘 활용하기 위한 기본 지식](https://bperhaps.tistory.com/entry/JAVA-enum%EC%9D%84-%EC%9E%98-%ED%99%9C%EC%9A%A9%ED%95%98%EA%B8%B0-%EC%9C%84%ED%95%9C-%EA%B8%B0%EB%B3%B8-%EC%A7%80%EC%8B%9D)

Reply:
`BinaryOperator` 처럼 뭔가 명시가 되지 않으면 좀 어색한 것 같아요! 😃 

Calculator 라는 이름은 경록님 의견에 동의합니다 👍 
연산기호 라는 표현할 수 있는 이름이 좋을 것 같네요!